app/models/tag.rb

```diff
@@ -201,4 +201,15 @@ def self.find_research_notes(tagnames, limit = 10)
         .order('node_revisions.timestamp DESC')
         .limit(limit)
   end
+
+  def followers_who_dont_follow_tags(tags)
+    tag_followers = self.subscriptions.map {|user| user.user_id}
+    users = User.find_all_by_id([tag_followers])
```

Comment:
Also here you can just do `User.find(uids)` -- or if you don't want to refactor DrupalUsers above, you can do `User.find(self.subscriptions.collect(&:user_id))` and it'll run on the whole array of ids.

I think `collect()` can make this code a lot more readable. It's hard to see what's going on! But now that you have a test, you can mess with it and always know if your code is correctly fulfilling the test :-)